File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> pref(n, 0);
        pref[0] = nums[0];
        for (int i = 1; i < n; i++) pref[i] = pref[i-1] + nums[i];
        int ans = 0;
        for (int i = 0; i < n - 1; i++) {
            int left = pref[i];
            int right = pref[n-1] - pref[i];
            if (left >= right) ans++;
        }
        return ans;
    }
};

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        n = len(nums)
        pref = [0] * n
        pref[0] = nums[0]
        for i in range(1, n):
            pref[i] = pref[i-1] + nums[i]
        ans = 0
        for i in range(0, n-1):
            left = pref[i]
            right = pref[n-1] - pref[i]
            if left >= right:
                ans += 1
        return ans